pkg/docker/Dockerfile.go1.22

```diff
@@ -1,3 +1,5 @@
+FROM rust:1.83.0-bookworm AS rust
```

Comment:
The `rust` image only maintains a single version at any given time, so this is only reasonable if it's going to be proactively kept up-to-date. :grimacing: